nautobot/ipam/models.py

```diff
@@ -531,8 +531,19 @@ def __init__(self, *args, **kwargs):
         prefix = kwargs.pop("prefix", None)
         self._location = kwargs.pop("location", None)
         super().__init__(*args, **kwargs)
+
         self._deconstruct_prefix(prefix)
 
+        self._parent = None
+        self._network = None
+        self._prefix_length = None
+        self._namespace = None
+
+        if self.pk:
```

Comment:
https://github.com/nautobot/nautobot/pull/6616#discussion_r1880232542

using self.pk at the __init__ level instead of self.present_in_database because currently thats the only way that works after testing several times.
self.present_in_database always returns False at __init__